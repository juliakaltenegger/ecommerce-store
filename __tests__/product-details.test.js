import getAllProducts, { getProductById } from "../product-details.js";

test("gets a product by id from the database", () => {
  expect(getProductById("2")).toEqual(
    getAllProducts().find(product => {
      return product.id === "2";
    }),
  );

  expect(getProductById("0")).toMatchSnapshot();
});

test("returns undefined when passed a non-id", () => {
  expect(getProductById("")).toBe(undefined);
  expect(getProductById(false)).toBe(undefined);
  expect(getProductById(null)).toBe(undefined);
});
