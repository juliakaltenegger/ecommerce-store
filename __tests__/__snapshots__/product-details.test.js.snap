// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets a product by id from the database 1`] = `
Object {
  "id": "0",
  "name": "Salsa Picante",
  "photo": "/static/salsa-picante.jpg",
  "price": "12,90",
  "spicyLevel": "medium",
}
`;
